src/libpmem2/map_posix.c

```diff
@@ -229,6 +248,11 @@ pmem2_map(const struct pmem2_config *cfg, struct pmem2_map **map_ptr)
 		return PMEM2_E_INVALID_FILE_HANDLE;
 	}
 
+	if ((int)cfg->requested_max_granularity == PMEM2_GRANULARITY_INVALID) {
+		ERR("the provided granularity value is invalid");
```

Comment:
this error message is very important to get right- probably >90% of first-time users will encounter it